src/garage/_dtypes.py

```diff
@@ -167,3 +166,91 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
                                actions, rewards, terminals, env_infos,
                                agent_infos, lengths)
+
+
+class TimeStep(
```

Comment:
i just realized here that validating every sample could have a big performance impact -- perhaps you can profile running `rollout` using this data structure versus a regular tuple?

Reply:
I'll run a profile. I'm writing a sampler to use with it right now that I'll include in this PR.